src/ASM/ASMs3Dmx.C

```diff
@@ -271,75 +271,79 @@ bool ASMs3Dmx::generateFEMTopology ()
   }
 #endif
 
-  nel = (m_basis[geoBasis-1]->numCoefs(0)-m_basis[geoBasis-1]->order(0)+1)*
-        (m_basis[geoBasis-1]->numCoefs(1)-m_basis[geoBasis-1]->order(1)+1)*
-        (m_basis[geoBasis-1]->numCoefs(2)-m_basis[geoBasis-1]->order(2)+1);
+  const Go::SplineVolume* itg = this->getBasis(ASM::INTEGRATION_BASIS);
+
+  nel = (itg->numCoefs(0) - itg->order(0) + 1) *
+        (itg->numCoefs(1) - itg->order(1) + 1) *
+        (itg->numCoefs(2) - itg->order(2) + 1);
 
   myMLGE.resize(nel,0);
   myMLGN.resize(nnod);
   myMNPC.resize(nel);
   myNodeInd.resize(nnod);
 
-  int i1, i2, i3, j1, j2, j3;
   size_t iel = 0, inod = 0;
   for (auto& it : m_basis)
-    for (i3 = 0; i3 < it->numCoefs(2); i3++)
-      for (i2 = 0; i2 < it->numCoefs(1); i2++)
-        for (i1 = 0; i1 < it->numCoefs(0); i1++)
+    for (int k = 0; k < it->numCoefs(2); k++)
+      for (int j = 0; j < it->numCoefs(1); j++)
+        for (int i = 0; i < it->numCoefs(0); i++)
         {
-          myNodeInd[inod].I = i1;
-          myNodeInd[inod].J = i2;
-          myNodeInd[inod].K = i3;
+          myNodeInd[inod].I = i;
+          myNodeInd[inod].J = j;
+          myNodeInd[inod].K = k;
           myMLGN[inod++]    = ++gNod;
         }
 
-  int firstItgElmNode = this->getFirstItgElmNode();
-  int nElmNode = std::accumulate(elem_size.begin(), elem_size.end(), 0);
+  const int firstItgElmNode = this->getFirstItgElmNode();
+  const size_t nElmNode = std::accumulate(elem_size.begin(), elem_size.end(), 0);
 
   // Create nodal connectivities for bases
+  auto&& enumerate_elem = [this,&iel](const Go::SplineVolume& spline,
```

Comment:
since `iel` is just a `size_t` and is not updated by the lambda, do we need to catch the reference here?

Reply:
yes, we do